testhelper/docker/resource/etcd/etcd.go

```diff
@@ -6,22 +6,49 @@ import (
 	"time"
 
 	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
 	etcd "go.etcd.io/etcd/client/v3"
 	"google.golang.org/grpc"
 
 	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource"
 )
 
 type Resource struct {
-	Client *etcd.Client
-	Hosts  []string
-	Port   int
+	Client           *etcd.Client
+	Hosts            []string
+	HostsSameNetwork []string
```

Comment:
Didn't get why it needed to be a slice. Can we simply have a single string variable for it something like `NetworkHost`?

Reply:
No reason, I copied from `Hosts` for consistency. I can't change `Hosts` since it'll be a breaking change so I kept this one like this too. I might be useful in case we do create more than one container like with the `kafka.Setup` so we might as well leave it I guess.